admin/tests-integration/azure-table.data.service.spec.js

```diff
@@ -0,0 +1,161 @@
+      expect(actual).toBeDefined()
+      expect(actual.partitionKey).toEqual(partitionKey)
+      expect(actual.rowKey).toEqual(rowKey)
+      expect(actual.customData).toEqual(customData)
+    })
+
+    it('throws an error when entity does not exist', async () => {
+      const tableName = getUniqueTableName()
+      await createTable(tableName)
+      const pk = uuid.v4()
+      const rk = uuid.v4()
+      try {
+        await sut.retrieveEntity(tableName, pk, rk)
+        fail('error should have been thrown')
+      } catch (error) {
+        expect(error.message).toEqual(`entity not found with PartitionKey:${pk} rowKey:${rk}`)
+      }
+    })
+  })
+
+  describe('clearTable', () => {
+    it('deletes all entries from the table', async () => {
+      const pk = uuid.v4()
+      const tableName = getUniqueTableName()
+      const client = TableClient.fromConnectionString(connectionString, tableName)
+      await createTable(tableName)
+      for (let index = 0; index < 5; index++) {
+        await client.createEntity({
+          partitionKey: pk,
+          rowKey: index.toString()
+        })
+      }
+      await sut.clearTable(tableName)
+      const entityIterator = client.listEntities()
+      let entityCount = 0
+      for await (const entity of entityIterator) {
+        entityCount++
+        fail(`found entity in table. pk:${entity.partitionKey} rk:${entity.rowKey}`)
+      }
+      expect(entityCount).toBe(0)
+    })
+  })
+
+  describe('createTables', () => {
+    it('creates specified tables in array', async () => {
+      const tableNames = [
+        getUniqueTableName(),
+        getUniqueTableName(),
+        getUniqueTableName()
+      ]
+      await sut.createTables(tableNames)
+      const client = TableServiceClient.fromConnectionString(connectionString)
+      const tableIterator = client.listTables()
+      const actualTables = []
+      for await (const table of tableIterator) {
+        actualTables.push(table.name)
+      }
+      const failures = []
+      for (let index = 0; index < tableNames.length; index++) {
+        const table = tableNames[index]
+        if (!RA.contained(actualTables, table)) {
+          failures.push(table)
+        }
+      }
+      if (failures.length > 0) {
+        let message = 'the following tables were not found...'
+        for (let index = 0; index < failures.length; index++) {
+          const table = failures[index]
+          message += `\n-${table}`
+        }
+        fail(message)
+      }
+    })
+
+    it('throws an error when one of the table names is invalid', async () => {
+      const randomPrefix = getUniqueTableName()
+      const tableNames = [
+        getUniqueTableName(),
+        `${randomPrefix}-Bad&chars^1`
+      ]
+      try {
+        await sut.createTables(tableNames)
+        fail('should have thrown an error due to hyphen in table name')
+      } catch (error) {
+        expect(error).toBeDefined()
```

Comment:
Jest has a better, more readable, idiom for async tests that are expected to throw:
`await expect(sut.createTables(tableNames)).rejects.toThrow(/some error message/)`

that would remove the entire try/catch block.

Reply:
updated both new tests that use try..catch